book/src/developer_guide/contributing.md

```diff
@@ -0,0 +1,65 @@
+# Contributor guide
+
+The `revive` compiler is an open source software project and we gladly accept quality contributions from anyone!
+
+## Getting started
+
+A quick reference on how to build the Solidity compiler is maintained in the projects [README.md](https://github.com/paritytech/revive?tab=readme-ov-file#building-from-source).
+
+### Using the `Makefile`
+
+The [Makefile](https://github.com/paritytech/revive/blob/main/Makefile) comprehensively encapsulates all development aspects of this codebase. It is kept concise and readable. Please read and use it! You'll learn for example:
+
+- How to build and install a `resolc` development version.
+- How to run tests and benchmarks.
+- How to cross-compile `resolc`.
+
+As a general rule-of-thumb: If `make test` runs fine locally, chances for green CI pipelines are good.
+
+### Codebase organization
+
+For the most parts, `revive` is a rather standard Rust workspace codebase. There are some non-Rust dependencies, which sometimes complicates things a little bit.
+
+#### The `crates/` dir
+
+All rust-crates live under the `crates/` directory. The workspace automatically consideres any crate found in there. If you need to add a new create, please implement it there.
+
+Compiler library crates should be named with the `revive-` prefix. The crate location doesn't need the prefix.
+
+#### Dependencies
+
+Dependencies should be added as workspace dependencies. Try to avoid pinning dependencies whenever possible. If possible to do so, add dev dependencies as `dev-dependencies` only.
```

Comment:
```suggestion
Dependencies should be added as workspace dependencies. Try to avoid pinning dependencies whenever possible. If possible, add dev dependencies as `dev-dependencies` only.
```